### ISSUE TITLE
[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] crash

### ISSUE THREAD

**Issue by jacob-sheldon** [SDWebImage/SDWebImage#3447]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 11.3
 SDWebImage Version      | 5.14.1
 Integration Method      | Swift Package Manager

### Issue Description and Steps

![image](https://user-images.githubusercontent.com/9928255/203309446-f68e7c21-7a08-4034-8fad-acb14b62fd55.png)

**Comment by dreampiggy** [comment]
Duplicated of https://github.com/SDWebImage/SDWebImage/issues/3433

**Comment by jacob-sheldon** [comment]
And I have encountered that crash about `queue = nil`, and after upgrade to 5.14.1, it don't appear anymore. but in these days this new one appears. Please check this new one.

**Comment by jacob-sheldon** [comment]
![image](https://user-images.githubusercontent.com/9928255/203508635-9a718ee0-1862-4d64-b4e9-356a93356f8a.png)
As the trace in RangerAPM, this line code is key, but I don't know why.

**Comment by dreampiggy** [comment]
https://github.com/SDWebImage/SDWebImage/issues/3447#issuecomment-1324744354

The line https://github.com/SDWebImage/SDWebImage/blob/5.14.1/SDWebImage/Core/SDWebImageDownloaderOperation.m#L614

and related function call does not use custom queue. It use the dispatchQueue.main. So I don't think it's a issue

And, from 5.14.1, there are no `NSOperationQueue.underlayingQueue` call in SDWebImage that class.

**Comment by dreampiggy** [comment]
If you think this is a new crash. Maybe more information should be provided, or any reproducable demo ? From the code logic here seems I can not get the reason.

**Comment by devssun** [comment]
Hello, I have a same issue on iOS 12.
- 95% of crash events for this issue occurred within the first 5 seconds of starting a user session.
- Crash events corresponding to this issue occurred on average 3 times per user.

I run on iOS 12.4 simulator but it works, so I can't attach Xcode crash reports.
Is this issue resolved at 5.14.1 version? please check.

### Issue Info

Info | Value
-- | --
Platform Name | ios, ipad
Platform Version | 12.5.6, 12.4.1, 12.5.5
SDWebImage Version | 5.14.0
Integration Method | spm
Xcode Version | Xcode 14.1
Repro rate | sometimes (2%)
Repro with our demo prj | not tested
Demo project link | n/a

<br class="Apple-interchange-newline">

### error log
```
Crashed: NSOperationQueue 0x282e8ebc0 (QOS: UNSPECIFIED)
0  libdispatch.dylib              0x62f8 dispatch_barrier_async$VARIANT$mp + 160
1  AppName                       0x32fabc -[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] + 614 (SDWebImageDownloaderOperation.m:614)
2  AppName                       0x32bbd8 -[SDWebImageDownloader URLSession:task:didCompleteWithError:] + 505 (SDWebImageDownloader.m:505)
3  FirebasePerformance            0x17298 __InstrumentURLSessionTaskDidCompleteWithError_block_invoke + 208
4  CFNetwork                      0x1df420 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.182 + 252
5  Foundation                     0x112cf4 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
6  Foundation                     0x1ed90 -[NSBlockOperation main] + 72
7  Foundation                     0x1e278 -[__NSOperationInternal _start:] + 740
8  Foundation                     0x114a78 __NSOQSchedule_f + 272
9  libdispatch.dylib              0x5fa38 _dispatch_call_block_and_release + 24
10 libdispatch.dylib              0x607d4 _dispatch_client_callout + 16
11 libdispatch.dylib              0x501c _dispatch_continuation_pop$VARIANT$mp + 412
12 libdispatch.dylib              0x46e0 _dispatch_async_redirect_invoke + 600
13 libdispatch.dylib              0x11030 _dispatch_root_queue_drain + 372
14 libdispatch.dylib              0x118d4 _dispatch_worker_thread2 + 128
15 libsystem_pthread.dylib        0xc1b4 _pthread_wqthread + 464
16 libsystem_pthread.dylib        0xecd4 start_wqthread + 4
```

**Comment by dreampiggy** [comment]
> Hello, I have a same issue on iOS 12.
> 
> * 95% of crash events for this issue occurred within the first 5 seconds of starting a user session.
> * Crash events corresponding to this issue occurred on average 3 times per user.
> 
> I run on iOS 12.4 simulator but it works, so I can't attach Xcode crash reports. Is this issue resolved at 5.14.1 version? please check.
> 
> ### Issue Info
> Info	Value
> Platform Name	ios, ipad
> Platform Version	12.5.6, 12.4.1, 12.5.5
> SDWebImage Version	5.14.0
> Integration Method	spm
> Xcode Version	Xcode 14.1
> Repro rate	sometimes (2%)
> Repro with our demo prj	not tested
> Demo project link	n/a
> ### error log
> ```
> Crashed: NSOperationQueue 0x282e8ebc0 (QOS: UNSPECIFIED)
> 0  libdispatch.dylib              0x62f8 dispatch_barrier_async$VARIANT$mp + 160
> 1  AppName                       0x32fabc -[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] + 614 (SDWebImageDownloaderOperation.m:614)
> 2  AppName                       0x32bbd8 -[SDWebImageDownloader URLSession:task:didCompleteWithError:] + 505 (SDWebImageDownloader.m:505)
> 3  FirebasePerformance            0x17298 __InstrumentURLSessionTaskDidCompleteWithError_block_invoke + 208
> 4  CFNetwork                      0x1df420 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.182 + 252
> 5  Foundation                     0x112cf4 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
> 6  Foundation                     0x1ed90 -[NSBlockOperation main] + 72
> 7  Foundation                     0x1e278 -[__NSOperationInternal _start:] + 740
> 8  Foundation                     0x114a78 __NSOQSchedule_f + 272
> 9  libdispatch.dylib              0x5fa38 _dispatch_call_block_and_release + 24
> 10 libdispatch.dylib              0x607d4 _dispatch_client_callout + 16
> 11 libdispatch.dylib              0x501c _dispatch_continuation_pop$VARIANT$mp + 412
> 12 libdispatch.dylib              0x46e0 _dispatch_async_redirect_invoke + 600
> 13 libdispatch.dylib              0x11030 _dispatch_root_queue_drain + 372
> 14 libdispatch.dylib              0x118d4 _dispatch_worker_thread2 + 128
> 15 libsystem_pthread.dylib        0xc1b4 _pthread_wqthread + 464
> 16 libsystem_pthread.dylib        0xecd4 start_wqthread + 4
> ```

I think you should at least using 5.14.1, which solve the similiar issue because dispatch_queue is nil https://github.com/SDWebImage/SDWebImage/releases/tag/5.14.1

**Comment by devssun** [comment]
Ok, thanks. I'll try 5.14.2. (latest version)
